usecases/decision_usecase.go

```diff
@@ -188,8 +189,10 @@ func (usecase *DecisionUsecase) CreateDecision(
 		return models.DecisionWithRuleExecutions{},
 			errors.Wrap(err, "error getting scenario")
 	}
-	if err := usecase.enforceSecurityScenario.ReadScenario(scenario); err != nil {
-		return models.DecisionWithRuleExecutions{}, err
+	if !skipScenarioPermissionForTransferCheck {
```

Comment:
I'm not all too happy with that, but it's a fair enough workaround for right now.
The reason I'm introducing this is that I don't want to give scenario_read permissions to the API keys used for transfer check (because we don't want clients to read the logic details we put in the rules).
A long term solution would be to factorize code differently here... but now I need to move on to the string fuzzy matching with higher priority.

Reply:
I'm not sure that works, the permission is held on the credentials which are in the context, at first look this seems unpractical (the usecase with its context is built at the API handler level)

Reply:
(and TBH I'm not 100% with how we handle that context injection, but we'll see about that issue in due time)